cirq/google/devices/known_devices.py

```diff
@@ -61,55 +65,77 @@ def create_device_proto_from_diagram(
         gate_sets: Optional[Iterable[
             serializable_gate_set.SerializableGateSet]] = None,
         durations_picos: Dict[str, int] = None,
+        out: Optional[device_pb2.DeviceSpecification] = None,
 ) -> device_pb2.DeviceSpecification:
-    """
-    Parse ASCIIart device layout into DeviceSpecification proto containing
-    information about qubits and targets.  Note that this does not populate
-    the gate set information of the proto.  This function also assumes that
-    only adjacent qubits can be targets.
+    """Parse ASCIIart device layout into DeviceSpecification proto.
 
-    Args:
-            ascii_grid: An ASCII version of the grid, see _parse_device for
-                    details
-            gate_set: A serializable gate_set object that is used to define
-                    the translation between gate ids and cirq Gate objects
-            durations_picos: A dictionary with keys of gate ids and values of
-                    the duration of that gate in picoseconds
+    This function also assumes that all pairs of adjacent qubits are valid
+    targets for two-qubit gates.
 
+    Args:
+        ascii_grid: ASCII version of the grid (see _parse_device for details).
+        gate_sets: Gate sets that define the translation between gate ids and
+            cirq Gate objects.
+        durations_picos: A map from gate ids to gate durations in picoseconds.
+        out: If given, populate this proto, otherwise create a new proto.
     """
-
     qubits, _ = _parse_device(ascii_grid)
-    spec = device_pb2.DeviceSpecification()
 
-    # Create valid qubit list
     qubit_set = frozenset(qubits)
-    spec.valid_qubits.extend([v2.qubit_to_proto_id(q) for q in qubits])
+    pairs: List[Tuple['cirq.Qid', 'cirq.Qid']] = []
+    for qubit in qubits:
+        for offset in [(0, 1), (1, 0)]:
```

Comment:
This duplicates a bit of logic already implemented in the `neighbors` methods of qubit classes. Could you use the method instead?

Reply:
We could, but would have to add extra logic here to avoid adding reversed pairs (e.g. want `(0, 0), (0, 1))` but not also `((0, 1), (0, 0))`). I can do that if you think it's worth i.

Reply:
nvm, went ahead and refactored to use the `neighbors` method.

Reply:
The code isn't using the `neighbors` method.

BTW: Another reason to use it is that (AFAIK) qubits may be instances of classes other than `GridQubit` and then adding a 2-tuple of ints won't work.

(Perhaps you forgot to push a commit?)

Reply:
Pushed. Note that this only works on the output of `_parse_device` which explicitly returns `cirq.GridQubit`, so adding tuples works. But in any case using `neighbors` is cleaner; thanks for the suggestion!